app/src/main/java/com/guzelgimadieva/tasky/authorization/ui/login/LoginScreenViewModel.kt

```diff
@@ -25,8 +37,24 @@ class LoginScreenViewModel @Inject constructor() : ViewModel() {
                 _loginState.update { _loginState.value.copy(passwordVisible = !_loginState.value.passwordVisible) }
             }
             is LoginEvent.Login -> {
-                // TODO: Implement login logic
+                viewModelScope.launch {
+                    sendLogin()
+                    //TODO use response and check if token is null
+                }
             }
         }
     }
+
+    private suspend fun sendLogin()  {
+        service.login(
+            LoginRequest(
+                _loginState.value.email,
+                _loginState.value.password
+            )
```

Comment:
Do you plan to work with a domain layer? If not, you can ignore this comment, but if so then you would couple the data layer to presentation here which isn't allowed in that type of architecture.